### PR TITLE
fixes #3876 - API find_resource by name even if name starts with integer

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by domcleal** [comment]
Lots of tests have failed, could you check them please?

**Comment by isratrade** [comment]
re-commited. For some reason, tests now pass locally with `=~` but didn't with `.match`

**Comment by domcleal** [comment]
Still failing unfortunately.

**Comment by isratrade** [comment]
@domcleal, this should work now.  I was testing locally the case of /:id or /:name, but the tests were using to_param which is /:id-:name.  All tests should pass now.

File: app/controllers/api/base_controller.rb [hunk 1]
```diff
@@ -130,7 +130,7 @@ def deny_access(details = nil)
 
     # possible keys that should be used to find the resource
     def resource_identifying_attributes
-      %w(id name)
+      %w(name id)
```
Comment (domcleal) [inline app/controllers/api/base_controller.rb:133]: This is overridden in both users controllers, they should be updated too.

Comment (thomasmckay) [inline app/controllers/api/base_controller.rb:133]: And in katello organizations controller. (Katello organizations use "label" as identifying attribute.) Please confirm everything works in foretello. Thanks!

Comment (isratrade) [inline app/controllers/api/base_controller.rb:133]: @domcleal, what's the plan per @thomasmckay's request. Is this going to be an automated process?

**Comment by isratrade** [comment]
re-commited with change to users controllers

**Comment by omaciel** [comment]
Looking forward to this :)

**Comment by isratrade** [comment]
@domcleal, can you commit this for @omaciel as the PR with friendly_id may need more testing.

**Comment by domcleal** [comment]
Thanks @isratrade, merged as 823d4e1 for Foreman 1.5.0.

File: app/controllers/api/base_controller.rb (hunk 1 above)
Comment (domcleal) [inline app/controllers/api/base_controller.rb:133]: If it needs changing in Katello too, a separate redmine will have to be filed there.

**Pull request closed without merging**